File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;


vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    //모든 작업이 끝날 때까지 반복
    int count = 0;
    int idx = 0;
    while(idx < progresses.size()){
            if(progresses[idx] >= 100){
                count++;
                idx++;
            }
            else{
                if(count != 0){
                    answer.push_back(count);
                    count = 0; //초기화
                }
                 // 하루가 지날 때마다 모든 progresses에 speeds 더하기
                for(int k = 0; k < progresses.size(); k++){
                    progresses[k] += speeds[k];
                }
            }
        }
    
    //남아있는 값 정리
    if(count != 0) answer.push_back(count);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.kt
class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        var answer = mutableListOf<Int>()
        val done = BooleanArray(progresses.size) //false로 초기화됨
        
        
        // 모든 작업이 끝날 때까지 
        while(!done[progresses.size -1]){
            // 하루의 작업 
            var count = 0
            for(i in 0 until progresses.size){
                if(!done[i]){
                    // 작업률 진행 
                    progresses[i] += speeds[i]

                    // 100 넘으면 배포 & 앞의꺼도 배포됐으면 
                    if((i == 0 || done[i-1]) && progresses[i] >= 100){
                        count++
                        done[i] = true
                    }
                }  
            }
            if(count > 0) answer.add(count)
        }
        
        return answer.toIntArray()
    }
}